master/internal/rm/kubernetesrm/pods.go

```diff
@@ -864,6 +907,133 @@ func (p *pods) podStatusCallback(s *actor.System, event watch.Event) {
 	}
 }
 
+var clusterID string
+
+func setClusterID(s string) {
+	if clusterID != "" {
+		panic(fmt.Sprintf("set cluster ID again new %s old %s", s, clusterID))
+	}
+	clusterID = s
+}
+
+func clusterIDNodeLabel() string {
+	return fmt.Sprintf("determined.ai/cluster-id-%s", clusterID)
+}
+
+const (
+	noExecuteNodeLabelValue  = "no-execute"
+	noScheduleNodeLabelValue = "no-schedule"
+)
+
+func (p *pods) enableNode(
+	ctx *actor.Context, nodeName string,
+) (*apiv1.EnableAgentResponse, error) {
+	patch := []byte(fmt.Sprintf(`{
+		"metadata": {
+			"labels": {
+				"%s": null
+			}
+		}
```

Comment:
thoughts on using `metav1.ObjectMeta` instead? and json marshalling it. does it work?

Reply:
That is a good idea

added it for enable but for disable since we need to use the JSON literal null to remove the label, and labels is a ```map[string]string```

I think its interesting how in almost every Go code base you can find things like this

Reply:
oh weird. setting it to "" just leaves it, doesn't remove it? yeah, this is grpc nulls all over again.